heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -97,54 +99,159 @@ const JEST_CONFIG_PACKAGE_FOLDER: string = path.dirname(require.resolve('jest-co
 /**
  * @internal
  */
-export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
-  public readonly pluginName: string = PLUGIN_NAME;
-  public readonly optionsSchema: JsonSchema = JsonSchema.fromFile(PLUGIN_SCHEMA_PATH);
+export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
+  private static _jestConfigurationFileLoader: ConfigurationFile<IHeftJestConfiguration> | undefined;
 
   /**
-   * Runs required setup before running Jest through the JestPlugin.
+   * Setup the hooks and custom CLI options for the Jest plugin.
+   *
+   * @override
    */
-  public static async _setupJestAsync(
-    scopedLogger: ScopedLogger,
+  public apply(
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    debugMode: boolean,
-    buildStageProperties: IBuildStageProperties
+    pluginOptions?: IJestPluginOptions
+  ): void {
+    // Flags
+    const detectOpenHandlesParameter: CommandLineFlagParameter = taskSession.parametersByLongName.get(
```

Comment:
Can we add a test that validates that all of the parameter long names from `heft-plugin.json` get retrieved, and no parameters that aren't in `heft-plugin.json` get retrieved?